libs/sdk-model/src/insight/index.ts

```diff
@@ -663,6 +672,49 @@ export function insightReduceItems<T extends IInsightDefinition>(
     } as T;
 }
 
+/**
+ * @public
+ */
+export type InsightDisplayFormUsage = {
+    /**
+     * References to display forms used for slicing and dicing the results.
+     */
+    inAttributes: ObjRef[];
+
+    /**
+     * References to display forms used to filter the entire insight.
+     */
+    inFilters: ObjRef[];
+
+    /**
+     * References to display forms used to filter particular measures.
+     */
+    inMeasureFilters: ObjRef[];
+};
+
+/**
+ * Gets references to all display forms used by the insight. The display forms may be used for slicing or dicing the
+ * data, for filtering the entire insight or for filtering just some measures.
+ *
+ * @param insight - insight to get the display form usage from
+ * @public
+ */
+export function insightDisplayFormUsage<T extends IInsightDefinition>(insight: T): InsightDisplayFormUsage {
+    invariant(insight, "insight must be specified");
+
+    return {
+        inAttributes: insightAttributes(insight).map(attributeDisplayFormRef),
+        inFilters: insightFilters(insight)
+            .filter(isAttributeFilter)
+            .map((attributeFilter) => filterObjRef(attributeFilter)),
+        inMeasureFilters: flatMap(insightMeasures(insight), (measure) => {
+            const filters = measureFilters(measure) ?? [];
+
+            return filters.filter(isAttributeFilter).map((attributeFilter) => filterObjRef(attributeFilter));
+        }),
```

Comment:
In all three places it might be useful to do some kind of uniq. Just because the code using this data will probably always treat these arrays like sets, not vectors, right?

Reply:
fixed